pages/battle.php

```diff
@@ -184,12 +184,14 @@ function processBattleFightEnd(BattleManager|BattleManagerV2 $battle, User $play
                     $player->daily_tasks->progressTask(DailyTask::ACTIVITY_DAILY_PVP, $rep_gained);
                 }
             }
-            $system->db->query("UPDATE `loot` SET 
-                `user_id` = {$player->user_id}, 
-                `battle_id` = NULL 
-              WHERE `battle_id` = {$player->battle_id} LIMIT " . ceil($battle->getTotalLoot() / 2));
+    
+            // Calculate the half of total loot
+            $half_loot = ceil($battle->getTotalLoot() / 2);
+    
+            // Update the loot table for the retreating player to lose half of their loot
+            $system->db->query("UPDATE `loot` SET `user_id` = {$battle->opponent->user_id}, `battle_id` = NULL WHERE `battle_id` = {$player->battle_id} LIMIT $half_loot");
```

Comment:
This query above looks like it will still assign half of the total loot from the battle to the retreating player, rather than half of the retreating player's loot. 

I think we needs to handle this in two steps
- Retreating player claims half their own loot, clearing the Battle ID from it
- Winning player claims all remaining loot for this battle (will be all their own loot + half the retreating players)

The retreating player's code branch will run first, so here in the winning branch we can just make this update all remaining loot to belong to the winning player and clear the battle ID from it.